src/libasr/asdl_cpp.py

```diff
@@ -1445,6 +1445,19 @@ def visitField(self, field, cons):
                     self.emit("} else {", 2)
                     self.emit(    's.append("()");', 3)
                     self.emit("}", 2)
+                elif field.name == "value":
+                    self.emit(  's.append(\"\\\"\");',2)
+                    self.emit(  "std::string str = std::string(x.m_value);",2)
+                    self.emit(  "for (size_t idx=0; idx < str.size(); idx++) {",2)
+                    self.emit(      "if (str[idx] == \'\\n\') {",3)
+                    self.emit(          "s += \"\\\\n\";",4)
+                    self.emit(      "} else if (str[idx] == '\\\\') {",3)
+                    self.emit(          "s += \"\\\\\\\\\";",4)
+                    self.emit(      "} else {",3)
+                    self.emit(          "s += str[idx];",4)
+                    self.emit(      "}",3)
+                    self.emit(  "}",2)
+                    self.emit(  "s.append(\"\\\"\");",2)
```

Comment:
I think we have to handle this in `semantics.h` itself.

Reply:
I have tried handling this in the semantics alone, but could not do it because that changes the output of the code. It changes the ```value``` in the ```ConstantStr``` node. If we change that, we'll have to handle this again in the backend (```python_ast_to_asr.cpp``` i guess) to get the same output as python. If that is what we need I'll try to do that myself first, otherwise you should take over this PR. Thank you